src/lib/quest.ts

```diff
@@ -0,0 +1,72 @@
+import { type Rating } from "./rating";
+
+export interface Quest<TConditionData, TState> {
+  type: QuestType;
+  conditionData: TConditionData;
+  createdAt: Date;
+  playerId: string;
+  state: TState;
+  description: () => string;
+  evaluate: (event: MatchWithPlayers) => QuestStatus;
+  reward: () => Rating;
+  penalty: () => Rating;
```

Comment:
Hvorfor er description, reward og penalty funktioner?

Reply:
Reward og Penalty er fordi jeg overvejer om der skulle sendes en `RatingSystemType` ind for at kunne give forskellige værdier. F.eks. en reward på `200` er ikke så meget når det er XP, men i Elo eller Openskill så det en rimelig significant reward :D 

Description er for at der potentielt kunne kaldes noget LLM for at generate en beskrivelse. Det var hvertfald den tanke jeg havde igår.

Reply:
Jeg havde nok egentligt tænkt at det var rating system der bestemte hvad der skulle ske ud fra en completed quest, frem for at questen selv er nødt til at kende til alle rating systemer.

Ang. description så skal den vel laves og fastlåses når questen oprettes, så du ikke den ene dag får én description og næste dag får en anden.

Reply:
Synes ikke det giver mening af have Reward / Penalty på questen på questen, nej